src/WebJobs.Script.WebHost/WebJobsApplicationBuilderExtension.cs

```diff
@@ -18,10 +18,8 @@ public static IApplicationBuilder UseWebJobsScriptHost(this IApplicationBuilder
 
         public static IApplicationBuilder UseWebJobsScriptHost(this IApplicationBuilder builder, IApplicationLifetime applicationLifetime, Action<WebJobsRouteBuilder> routes)
         {
-            WebScriptHostManager hostManager = builder.ApplicationServices.GetService(typeof(WebScriptHostManager)) as WebScriptHostManager;
-
+            builder.UseMiddleware<FunctionInvocationMiddleware>();
```

Comment:
Can you elaborate on why this is the first middleware? I would have thought it'd be sitting after the routing one.

Reply:
Once handled by the router, other middleware in the pipeline will not be executed, so this is registered first. You'll notice the actual middleware implementation handles the logic after executing the rest of the pipeline.